src/cite.cpp

```diff
@@ -152,46 +152,49 @@ void CitationManager::generatePage()
             int k=line.find("}",i);
             if (j!=-1 && k!=-1)
             {
-              QCString label = line.mid((uint)(j+1),(uint)(k-j-1));
+              QCString crossrefName = line.mid((uint)(j+1),(uint)(k-j-1));
               // check if the reference with the cross reference is used
               // insert cross refererence when cross reference has not yet been added.
-              if ((p->entries.find(label1.data())!=p->entries.end()) &&
-                  (p->entries.find(label.data())==p->entries.end())) // not found yet
+              if ((p->entries.find(citeName.data())!=p->entries.end()) &&
+                  (p->entries.find(crossrefName.data())==p->entries.end())) // not found yet
               {
-                insert(label);
+                insert(crossrefName);
               }
             }
           }
           else if (line.stripWhiteSpace().startsWith("@"))
           {
             // assumption entry like: "@book { name," or "@book { name" (spaces optional)
-            int j=line.find("{",0);
+            int j=line.find("{");
             // when no {, go hunting for it
             while (j==-1 && (s=input.find('\n',pos))!=-1)
             {
               line = input.mid((uint)pos,(uint)(s-pos));
-              j=line.find("{",0);
+              j=line.find("{");
               pos=s+1;
             }
             // search for the name
-            label1 = "";
-            while (label1.isEmpty())
+            citeName = "";
+            if (j != -1) // to prevent something like "@manual ," and no { found
             {
-              int k=line.find(",",j);
-              if (k != -1)
+              while (s != -1 && citeName.isEmpty())
               {
-                label1 = line.mid((uint)(j+1),(uint)(k-j-1));
-              }
-              else
-              {
-                label1 = line.mid((uint)(j+1));
-              }
-              label1 = label1.stripWhiteSpace();
-              j = -1;
-              if (label1.isEmpty() && (s=input.find('\n',pos))!=-1)
-              {
-                line = input.mid((uint)pos,(uint)(s-pos));
-                pos=s+1;
+                int k=line.find(",",j);
+                if (k != -1)
+                {
+                  citeName = line.mid((uint)(j+1),(uint)(k-j-1));
+                }
+                else
+                {
+                  citeName = line.mid((uint)(j+1));
+                }
+                citeName = citeName.stripWhiteSpace();
+                j = -1;
```

Comment:
after this assignment the while loop at line 180 will do a find with position variable j==-1 on line 182. That doesn't look correct. May the `s!=-1` guard should be `j`? code is still a bit cryptic for my understanding.

Reply:
This index looks indeed a bit suspicious, but the guard definitely should not be based on `j` as we want to detect the end of the "file" / "buffer" in the while loop.
I have to check some "nasty" examples.

Reply:
Indeed there was a regression regarding the `,`. Added update and example (in separate comment, not allowed in review).